src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -218,12 +218,12 @@ void GenerateImplementationFor(Type interfaceType)
         {
             if (!interfaceType.IsVisible)
             {
-                throw new Exception($"Can only generate a concrete implementation for '{interfaceType}' if '{interfaceType}' is public.");
+                throw new Exception($"Cannot generate a concrete implementation for '{interfaceType}' because it is not public. Ensure that all interfaces used as messages are public.");
             }
 
             if (interfaceType.GetMethods().Any(mi => !(mi.IsSpecialName && (mi.Name.StartsWith("set_") || mi.Name.StartsWith("get_")))))
             {
-                throw new Exception($"Can only generate a concrete implementation for '{interfaceType.Name}' because the interface contains methods. Ensure interface messages do not contain methods.");
+                throw new Exception($"Cannot generate a concrete implementation for '{interfaceType.Name}' because it contains methods. Ensure that all interfaces used as messages do not contain methods.");
```

Comment:
writing the unit tests, I noticed that we're using `{interfaceType.Name}` here but just `{interfaceType}` for the private interface exception message on line 221. Should we clarify this and probably use `{interfaceType.Name.FullName}` in both cases?

Reply:
Yeah, that make sense. I was focused on the error text and didn't notice the difference between the two.

Reply:
@bording will you make the mentioned change?

Reply:
Done!